ddtrace/contrib/langchain/patch.py

```diff
@@ -348,6 +368,14 @@ def traced_chat_model_generate(langchain, pin, func, instance, args, kwargs):
         integration.metric(span, "incr", "request.error", 1)
         raise
     finally:
+        if integration.is_pc_sampled_llmobs(span):
```

Comment:
Side note: I found a bug in `is_pc_sampled_llmobs`. This method does not always return the correct sampling decision. `BaseLLMIntegration._llmobs_pc_sampler.sample(span)` only returns a boolean, it does not make a sampling decision. Sampling decisions are recorded by setting the `_sampling_priority_v1` tag on a span. Right now all LLM spans are being sampled by the global tracer and this tracer could be using a different sampler (ex: Rate by Service Sampler).

I am not sure if this is a bug in LLM integrations or the RateSampler.

@ZStriker19 should the RateSampler make a sampling decision?

cc: @Yun-Kim

Reply:
It can make a decision, but it doesn't assign tags. Up for discussion if it should, but we'd have to rework some other code.

Reply:
For what it's worth, `is_pc_sampled_llmobs()` isn't meant to interfere in any sampling decisions for APM spans submitted to the trace agent, it's designed for determining whether or not to create and submit an LLMObs span event from the corresponding APM span. It should be independent from APM sampling in general (we are planning on becoming less dependent on the RateSampler for this)

Reply:
Will resolve this for now and we can have this as an offline discussion.